### PR TITLE
tests: add can create filter test

### PR DESCRIPTION
**Pull request by meenal06**
Signed-off-by: Meenal Trivedi <meenaltrivedi6102@gmail.com>

### CODE REVIEW

**Comment by meenal06** [comment]
Hi @Kegsay , I am unable to debug the failure cause, is it due to the line `map[string]string` rather than `map[string]int` due to type conversion in python?

**Comment by Kegsay** [comment]
Correct, it shouldn't be a string, use int instead.

File: tests/sync_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+func TestSyncFilter(t *testing.T) {
+	deployment := Deploy(t, "sync_filter", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
+	// sytest: Can create filter
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"room": map[string]interface{}{
+			"timeline": map[string]string{
+				"limit": "10",
```
Comment (Kegsay) [inline tests/sync_filter_test.go:20]:
```go
"timeline": map[string]int{
	"limit": 10,
},
```

File: tests/sync_filter_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"encoding/json"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestSyncFilter(t *testing.T) {
+	deployment := Deploy(t, "sync_filter", b.BlueprintAlice)
```
Comment (Kegsay) [inline tests/sync_filter_test.go:13]: Please remove `"sync_filter"` from this as it is no longer required as of https://github.com/matrix-org/complement/pull/113

**Comment by meenal06** [comment]
@Kegsay Done the changes, & woah CI was super fast in this! 🚀

**Review by Kegsay — APPROVED** [review]

**Pull request merged**